src/plugins/embeddable/public/lib/actions/edit_panel_action.ts

```diff
@@ -94,6 +95,12 @@ export class EditPanelAction implements Action<ActionContext> {
       const oldExplicitInput = embeddable.getExplicitInput();
       const newExplicitInput = await factory.getExplicitInput(oldExplicitInput, embeddable.parent);
       embeddable.parent?.replaceEmbeddable(embeddable.id, newExplicitInput);
+      if (
+        isReferenceOrValueEmbeddable(embeddable) &&
+        embeddable.inputIsRefType(embeddable.getInput())
+      ) {
+        embeddable.reload();
+      }
```

Comment:
We can make this better by making changes to the `Container.replaceEmbeddable` method to be functionally similar to the replacePanel function in `dashboard/public/dashboard_container/embeddable/api/panel_management.ts`.

Reply:
Draft PR to address this: https://github.com/nickpeihl/kibana/pull/2